src/mbgl/renderer/render_layer.hpp

```diff
@@ -69,16 +79,23 @@ class RenderLayer {
 
     virtual std::unique_ptr<Bucket> createBucket(const BucketParameters&, const std::vector<const RenderLayer*>&) const = 0;
 
+    void setRenderTiles(std::vector<std::reference_wrapper<RenderTile>>);
     // Private implementation
     Immutable<style::Layer::Impl> baseImpl;
     void setImpl(Immutable<style::Layer::Impl>);
 
+    virtual const style::Layer::Impl& impl() const = 0;
```

Comment:
But both of those methods only need a `style::Layer::Impl&`, not a derived `Layer::Impl`.